contracts/l2/L2PublicResolver.sol

```diff
@@ -17,35 +16,75 @@ import {NameResolver} from "./profiles/name/NameResolver.sol";
  * ensure that a link between the owner and the node is established.
  @dev Find the original contract here: https://github.com/ensdomains/resolvers/blob/master/contracts/PublicResolver.sol
  */
-contract L2PublicResolver is Multicallable, AddrResolver, TextResolver, ABIResolver, ContentHashResolver, DNSResolver, NameResolver {
+contract L2PublicResolver is Multicallable, AddrResolver, TextResolver, ABIResolver, ContentHashResolver, DNSResolver {
     /**
-     * @dev Checks whether the contract supports a specific interface by its identifier.
-     * @param interfaceID The identifier of the interface to check, represented as a bytes4 value.
-     * @return A boolean value indicating whether the contract supports the given interface.
-     *
-     * This function allows the contract to be queried for the support of specific interfaces defined
-     * by their unique four-byte identifier. It serves as an override for the same function in several
-     * parent contracts (Multicallable, AddrResolver, TextResolver, ABIResolver, ContentHashResolver, DNSResolver, NameResolver).
-     *
-     * The contract inheriting this function should implement the `supportsInterface` function
-     * in all the parent contracts and return true if any of them supports the specified interface.
-     * Otherwise, it returns false if none of the parent contracts support the interface.
-     *
-     * Example Usage:
-     * ```
-     * // Assuming `contractInstance` is an instance of the contract containing this function
-     * bytes4 interfaceId =  L2PublicResolver.supportsInterface.selector
-     * bool isSupported = contractInstance.supportsInterface(interfaceId);
-     * ```
+     * A mapping of delegates. A delegate that is authorised for a context
+     * for a name may make changes to the record.
+     * (context, name, delegate) => approved
+     */
+    mapping(bytes => mapping(bytes => mapping(address => bool))) private _approvals;
+
+    event Approved(bytes context, bytes name, address indexed delegate, bool indexed approved);
+
+    /**
+     * @dev Grants or revokes approval status for a delegate.
+     * @param name The name for which the approval status is being set.
+     * @param delegate The address of the delegate whose status is being set.
+     * @param approved A boolean indicating whether the delegate's approval status is being granted or revoked.
+     * @notice Only the owner of this approval can perform this action.
+     * @notice Setting delegate status for self is not allowed.
+     */
+    function approve(bytes calldata name, address delegate, bool approved) external {
+        require(msg.sender != delegate, "Setting delegate status for self");
```

Comment:
We could support ERC1271 here and allow this to take a signature that the contract verifies. If we don't, smart contract wallets won't be able to approve